packages/admin/admin/src/stack/breadcrumbs/utils.tsx

```diff
@@ -0,0 +1,89 @@
+import { ClassKeyOfStyles, ClassNameMap } from "@mui/styles";
+import * as React from "react";
+
+import { BreadcrumbItem } from "../Stack";
+import { BreadcrumbsEntry } from "./BreadcrumbsEntry";
+import { BreadcrumbsOverflow } from "./BreadcrumbsOverflow";
+import { styles } from "./StackBreadcrumbs.styles";
+
+export const getElementOuterWidth = (element: Element): number =>
+    element.clientWidth + parseFloat(getComputedStyle(element).marginLeft) + parseFloat(getComputedStyle(element).marginRight);
+
+const NUMBER_OF_ITEMS_BEFORE_OVERFLOW_MENU = 1;
+
+const useNumberOfItemsToBeHidden = (
+    items: BreadcrumbItem[],
+    containerWidth: number,
+    showBackButton: boolean,
+    itemWidths: number[] | undefined,
+): number | undefined => {
+    const [numberOfItemsToBeHidden, setNumberOfItemsToBeHidden] = React.useState<number | undefined>();
+
+    React.useEffect(() => {
+        let allVisibleItemsFitIntoContainer = false;
+        let newNumberOfItemsToBeHidden = 0;
+
+        const minimumNumberOfVisibleItems = 3; // First item, overflow menu & last item
+        const maximumNumberOfHiddenItems = items.length - minimumNumberOfVisibleItems + 1;
+
+        while (!allVisibleItemsFitIntoContainer && newNumberOfItemsToBeHidden < maximumNumberOfHiddenItems) {
+            let totalWidthOfVisibleItems = 0;
+
+            itemWidths?.forEach((itemWidth, index) => {
+                const isOverflowMenu = index === NUMBER_OF_ITEMS_BEFORE_OVERFLOW_MENU;
+                const overflowMenuWillBeShown = newNumberOfItemsToBeHidden > 0;
+                const linkItemWillBeShown =
+                    index === 0 || // Always show first item
+                    index === itemWidths.length - 1 || // Alawys show last item
+                    index > newNumberOfItemsToBeHidden + minimumNumberOfVisibleItems - 2;
+
+                if ((isOverflowMenu && overflowMenuWillBeShown) || (!isOverflowMenu && linkItemWillBeShown)) {
+                    totalWidthOfVisibleItems += itemWidth;
+                }
+            });
+
+            if (totalWidthOfVisibleItems > containerWidth) {
+                newNumberOfItemsToBeHidden++;
+            } else {
+                allVisibleItemsFitIntoContainer = true;
+            }
+        }
+
+        setNumberOfItemsToBeHidden(newNumberOfItemsToBeHidden);
+    }, [items, itemWidths, containerWidth, showBackButton, numberOfItemsToBeHidden]);
```

Comment:
Why do you have the dependency on `numberOfItemsToBeHidden`. I believe this may cause an effect loop